### PR TITLE
Fix Windows coreclr.dll size regression

### LINKED ISSUE #39599: coreclr.dll size regressed ~200kB due to constexpr changes
https://github.com/dotnet/runtime/pull/39532 introduce unexpected ~200kB size regression in coreclr.dll size on Windows x64:

commit a9b0bebdbb3b7fc50f295bb26acaa39b045634ae

5,028,352 coreclr.dll

commit 902645707215bcd0c22900d0a9ba4d01ca1a2fbe

5,209,600 coreclr.dll

### PR DESCRIPTION
**Pull request by janvorli**
A recent change has replaced usage of SELECTANY by constexpr to
significantly reduce size of coreclr binaries on Unix. But it has
resulted in about 180kB regression in coreclr.dll size on Windows.

This change gets rid of the constexpr at places that can be reasonably
moved to having declaration in header and definition in a .cpp file.
I've also found a couple of things that are not used anymore, so I have
deleted them.

The only thing where I've moved back to using SELECTANY on Windows is
the DbgIPCEventTypeNames in dbgipcevents.h. That table and the related
GetEventType / GetEventName methods are used in both mscordbi and
coreclr and sharing a .cpp source with the data between those two
libraries was ugly and problematic.

Close #39599

### CODE REVIEW

**Comment by janvorli** [comment]
cc: @dotnet/jit-contrib for the jit changes.

File: src/coreclr/src/gcinfo/simplerhash.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+#include "simplerhash.h"
+
+// Table of primes and their magic-number-divide constant.
+// For more info see the book "Hacker's Delight" chapter 10.9 "Unsigned Division by Divisors >= 1"
+// These were selected by looking for primes, each roughly twice as big as the next, having
+// 32-bit magic numbers, (because the algorithm for using 33-bit magic numbers is slightly slower).
+//
+
+PrimeInfo primeInfo[] =
```
Comment (jkotas) [inline src/coreclr/src/gcinfo/simplerhash.cpp:12]: Nit: `const` ?

File: src/coreclr/src/debug/inc/dbgipcevents.h [hunk 2]
```diff
@@ -984,7 +984,12 @@ struct MSLAYOUT IPCEventTypeNameMapping
             const char *            eventName;
 };
 
-constexpr IPCEventTypeNameMapping DbgIPCEventTypeNames[] =
+#ifdef TARGET_WINDOWS
```
Comment (jkotas) [inline src/coreclr/src/debug/inc/dbgipcevents.h:987]: Nit: `#if defined(_MSC_VER)` may be more appropriate

**Review by jkotas — APPROVED** [review]
LGTM otherwise. Thanks!

**Review by CarolEidt — APPROVED** [review]
LGTM

**Pull request merged**